### PR TITLE
[ci] Speed up CI with `macos-14`

### PR DESCRIPTION
**Pull request by styfle**
- Before: [5m 38s](https://github.com/vercel/vercel/actions/runs/7831733256/job/21368857518?pr=11131)
- After: [1m 23s](https://github.com/vercel/vercel/actions/runs/7836156637/job/21383185433?pr=11133)

Learn more: https://github.blog/changelog/2024-01-30-github-actions-macos-14-sonoma-is-now-available/

### CODE REVIEW

**Review by trek — DISMISSED** [review]

**Comment by styfle** [comment]
I think this might be an actual bug but I don't have time to look into it:

https://github.com/vercel/vercel/actions/runs/7846798470/job/21414416724?pr=11133#step:10:1979

```
vercel:test:     ✕ should detect update command (46 ms)
vercel:test: 
vercel:test:   ● getUpdateCommand › should detect update command
vercel:test: 
vercel:test:     expect(received).toEqual(expected) // deep equality
vercel:test: 
vercel:test:     Expected: "pnpm i vercel@latest"
vercel:test:     Received: "pnpm i -g vercel@latest"
```

Perhaps the new test runner includes pnpm as a global or maybe we are not detecting it properly.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ jobs:
           node-version: ${{ matrix.nodeVersion || env.NODE_VERSION }}
 
       - name: Install Hugo
-        if: matrix.runner == 'macos-latest'
+        if: matrix.runner == 'macos-14'
         run: curl -L -O https://github.com/gohugoio/hugo/releases/download/v0.56.0/hugo_0.56.0_macOS-64bit.tar.gz && tar -xzf hugo_0.56.0_macOS-64bit.tar.gz && mv ./hugo packages/cli/test/dev/fixtures/08-hugo/
```
Comment (styfle) [inline .github/workflows/test.yml:71]: I don't see hugo tests failing. The failing test seems to indicate that pnpm might be globally installed now

https://github.com/vercel/vercel/actions/runs/7907637325/job/21585186350?pr=11133#step:10:2003

**Comment by styfle** [comment]
Looks like this is blocked by https://github.com/vercel/vercel/pull/11405

**Comment by styfle** [comment]
@TooTallNate @EndangeredMassa @trek CI is finally green ✅ 

Please take another look.

**Review by EndangeredMassa — APPROVED** [review]

**Comment by EndangeredMassa** [comment]
Thanks @styfle @TooTallNate!

**Pull request merged**